tests/boards/intel_adsp/hda_log/src/logger.c

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
```

Comment:
Similar content to elsewhere: put the test code in a separate patch from the driver.  Especially with this feature, the protocol needed to do the management is wildly different from what SOF will use.